File: weather-dashboard/backend/weather.py
# weather.py
import os
import requests
from dotenv import load_dotenv
from pprint import pprint
load_dotenv()
API_KEY = os.getenv("OPENWEATHER_API_KEY")
unit = os.getenv("DEFAULT_UNIT")

def get_weather(city: str):
    if not city:
        raise Exception("City name is required")
    
    response = requests.get(
        f"http://api.openweathermap.org/data/2.5/weather?q={city}&appid={API_KEY}&units=metric"
    )
    
    if response.status_code == 200:
        data = response.json()
        # pprint(data)
        return {
            "city": data["name"],
            "temp": data["main"]["temp"],
            "feels_like": data["main"]["feels_like"],
            "humidity": data["main"]["humidity"],
            "wind": data["wind"]["speed"],
            "visibility": data["visibility"],
            "weather": data["weather"][0]["description"]
        }
    else:
        raise Exception("Error fetching weather", response.status_code, response.text)

File: weather-dashboard/weather-ui/lib/weather.ts
export interface WeatherData {
  city: string;
  temperature: number;
  description: string;
  humidity: number;
  windSpeed: number;
  visibility: number;
  feelsLike: number;
}

export async function getWeather(city: string): Promise<WeatherData> {
  const response = await fetch(
    `http://localhost:8000/weather?city=${encodeURIComponent(city)}`
  );

  if (!response.ok) {
    throw new Error("Failed to fetch weather data");
  }

  const data = await response.json();
  const visibility_m = data.visibility ?? 0;
  const visibility_km = visibility_m / 1000;

  return {
    city: data.city,
    temperature: data.temperature,
    description: data.weather,
    humidity: data.humidity,
    windSpeed: data.wind,
    visibility: visibility_km,
    feelsLike: data.feels_like, // match your backend JSON keys
  };
}
